codex/slots/sampler/utils.nim

```diff
@@ -47,10 +38,8 @@ func floorLog2*(x : int) : int =
   return k
 
 func ceilingLog2*(x : int) : int =
-  if (x == 0):
-    return -1
-  else:
-    return (floorLog2(x-1) + 1)
+  doAssert ( x > 0 )
```

Comment:
Might be a good idea to provide a reason string here so if we do encounter this defect, we can see something that’ll give us a clue as to what went wrong. Otherwise, we might not even get a stack trace to help.

Reply:
Same with line 32